Task/680/sync settings env

NewtonLC opened the pull request:
# Description

Added a script `scripts/check_env_sync.py` that:
- Reports Pydantic Validation Error, which occurs when `Settings` is not in sync with the `ENV_FILE` in `backend/api/config.py`
- Parses the `Settings` class in `backend/api/config.py` to extract all defined environment variable names.
- Reads `.env.example` and compares variable keys.
- Reports any variables missing from `.env.example`

Added a pre-commit hook to ensure that this script is run when a developer commits changes to `backend/api/config.py`.

Closes #680 

## Type of changes
- ( ) Bugfix
- (X) Chore
- ( ) New Feature

## Testing
- ( ) I added automated tests
- ( ) I think tests are unnecessary

## How to test

Edit `backend/api/config.py` and `.env`:
- Add new fields to the `Settings` class in `backend/api/config.py`
- Make sure that field also exists in `.env`, to avoid a pydantic validation error
- Alternatively, comment out fields in the `.env.example` file
  - Make sure to edit `backend/api/config.py` with a comment, print statement, etc. as the hook only runs if this file is changed.

`git add` the changed files, then `git commit` with a test message. 

You should see a message like this in the terminal, at the end of the pre-commit hook messages.
```
Check .env.example matches Settings......................................Failed
- hook id: check-env-sync
- exit code: 1

Missing environment variables in .env.example:
  - TEST_VAR
```

## Clean commits
- (X) I plan to Squash and Merge
- ( ) My commit history is clean¹
  ¹ [_described here_](./README.md#pull-requests)

NewtonLC committed:
Enforce Synchronization between Settings Class and .env.example

NewtonLC committed:
Enforce Synchronization between Settings Class and .env.example

NewtonLC committed:
Merge branch 'task/680/sync-settings-env' of https://github.com/sfbrigade/datasci-earthquake into task/680/sync-settings-env

NewtonLC committed:
Merge branch 'task/680/sync-settings-env' of https://github.com/sfbrigade/datasci-earthquake into task/680/sync-settings-env

NewtonLC committed:
Merge branch 'task/680/sync-settings-env' of https://github.com/sfbrigade/datasci-earthquake into task/680/sync-settings-env

NewtonLC committed:
Implemented gemini-code-assist suggestions, made code cleaner

NewtonLC committed:
Merge branch 'develop' into task/680/sync-settings-env

agennadi reviewed (approved):
Looks good, just a few nits: 1) you check that `Settings` exists but don't check if `.env.example` exists. 2) `parse_env_example()` doesn't check lines that aren't comments and don't contain `=`

NewtonLC committed:
Implemented Code Review Suggestions
Added explicit check for .env.example
Added check for malformed lines in .env.example
Added always_run check in pre-commit hook

NewtonLC committed:
Merge branch 'task/680/sync-settings-env' of https://github.com/sfbrigade/datasci-earthquake into task/680/sync-settings-env

NewtonLC commented:
Thanks for the review! I added your suggestions, so the script should be able to check whether `.env.example` exists in the pre-commit hook as well as any malformed lines.

NewtonLC committed:
Merge branch 'develop' into task/680/sync-settings-env
